crates/transaction-emitter-lib/src/emit.rs

```diff
@@ -642,21 +622,58 @@ impl<'t> TxnEmitter<'t> {
         let stop = Arc::new(AtomicBool::new(false));
         let stats = Arc::new(StatsAccumulator::default());
         let tokio_handle = Handle::current();
+        let mut creator_account = LocalAccount::generate(&mut self.rng);
+        let collection_name = "collection name".to_owned().into_bytes();
+        let token_name = "token name".to_owned().into_bytes();
+        if let TransactionType::NftMint = req.transaction_type {
+            initialize_nft_collection(
+                req.rest_clients[0].clone(),
+                self.root_account,
+                &mut creator_account,
+                &self.txn_factory,
+                &collection_name,
+                &token_name,
+            )
+            .await;
+        };
+        let nft_creator_account = Arc::new(creator_account);
```

Comment:
This seems a bit messy. 
One alternative would be to have TransactionGeneratorCreator for each of the types, that initializes all of this in new or newAndInitialize method, and has .createTxnGenerator() , such that 647-668 become just let txn_generator = tnx_generator_creator.createTxnGenerator(), and you have only single match/if on the transaction_type.

Reply:
Good point - I was not a fan of this as well. Thanks for the suggestion - fixed it.